diff --git a/node_modules/webpack-module-nomodule-plugin/src/index.js b/node_modules/webpack-module-nomodule-plugin/src/index.js
index c4e5034..ac4177f 100644
--- a/node_modules/webpack-module-nomodule-plugin/src/index.js
+++ b/node_modules/webpack-module-nomodule-plugin/src/index.js
@@ -129,7 +129,7 @@ class HtmlWebpackEsmodulesPlugin {
     scripts.forEach(s => {
       body.splice(body.indexOf(s), 1);
     })
-
+    debugger;
     modernScripts.forEach(modernScript => {
       head.push({ tagName: 'link', attributes: { rel: 'modulepreload', href: modernScript.attributes.src } });
     });
@@ -140,15 +140,31 @@ class HtmlWebpackEsmodulesPlugin {
   }
 
   sizeEfficient(existingAssets, body) {
+    const isModern = this.mode === 'modern';
+    const legacyScripts = (isModern ? existingAssets : body).filter(tag => tag.tagName === 'script' && tag.attributes.type !== 'module');
+    const modernScripts = (isModern ? body : existingAssets).filter(tag => tag.tagName === 'script' && tag.attributes.type === 'module');
+    const scripts = body.filter(tag => tag.tagName === 'script');
+    scripts.forEach(s => {
+      body.splice(body.indexOf(s), 1);
+    })
+    const legacyInitScript = {
+      tagName: 'script',
+      closeTag: true,
+      innerHTML: `noModules()&&document.write('${legacyScripts.map(script => `<scr'+'ipt type=\"text/javascript\" src=\"${script.attributes.src}\"></scr'+'ipt>`).join('')}');`,
+    };
+    const modernInitScript = {
+      tagName: 'script',
+      closeTag: true,
+      innerHTML: `!noModules()&&document.write('${modernScripts.map(script => `<scr'+'ipt type=\"module\" src=\"${script.attributes.src}\"></scr'+'ipt>`).join('')}');`,
+    };
     const safariFixScriptTag = {
       tagName: 'script',
       closeTag: true,
       innerHTML: safariFixScript,
     }
-
+    body.push(legacyInitScript);
     body.push(safariFixScriptTag);
-    body.push(...existingAssets);
-
+    body.push(modernInitScript);
     // Make our array look like [modern, script, legacy]
     if (this.mode === 'legacy') {
       body.reverse();
